src/angleaxis_types.jl

```diff
@@ -110,7 +110,7 @@ function Base.:*(aa::AngleAxis, v::StaticVector)
     st, ct = sincos(aa.theta)
     w_cross_pt = cross(w, v)
     m = dot(v, w) * (one(w_cross_pt[1]) - ct)
-    T = promote_type(eltype(aa), eltype(v))
+    T = Base.promote_op(*, eltype(aa), eltype(v))
```

Comment:
I don't fully understand the practical reason why `Base.promote_op` is avoided, but `Base.promote_op` is already used in other places in Rotations. jl like in this PR, so I will approve this PR.